core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -894,27 +919,36 @@ protected ComputerLauncher grabLauncher(Node node) {
     /**
      * Get the agent version
      */
+    @CheckReturnValue
     public String getSlaveVersion() throws IOException, InterruptedException {
+        if (channel == null) {
+            return "Unknown (agent is offline)";
```

Comment:
IOException may be better here, because the downstream code may try to parse the version and break